### ISSUE TITLE
Support userdata on QEMU PPC64

### ISSUE THREAD

**Issue by bgilbert** [coreos/ignition#666]
# Feature Request #

## Environment ##

QEMU ppc64

## Desired Feature ##

Support reading an Ignition config from userdata.  `fw_cfg` is not supported on ppc64.

## Other Information ##

SMBIOS OEM strings are also not supported; see https://github.com/coreos/ignition/issues/656#issuecomment-441898962.

**Comment by bgilbert** [comment]
Ideally QEMU and libvirt would gain support for a generic userdata mechanism that works on all CPU architectures; see https://github.com/coreos/ignition/issues/656#issuecomment-442160199.

**Comment by jcajka** [comment]
Could we in the mean time drop the hard requirement on the qemu_fw_cfg as it break the coreos assembler on platforms that are not supported by it. And from my understanding it is not only way how to interact with ignition.

**Comment by dustymabe** [comment]
> Could we in the mean time drop the hard requirement on the qemu_fw_cfg as it break the coreos assembler on platforms that are not supported by it. 

What specific error are you seeing? 

> And from my understanding it is not only way how to interact with ignition, right?

It is on the `qemu` (local qemu/libvirt) platform.

**Comment by jcajka** [comment]
@dustymabe respective kernel module doesn't exits on those platforms so dracut during initramfs always errorout

**Comment by dustymabe** [comment]
could you try to provide `ignition.config.url=http://example.com/ignition.ign` on the kernel command line and see if it still errors out?

**Comment by jcajka** [comment]
It is at build time of artifacts.

**Comment by ajeddeloh** [comment]
Ignition itself doesn't require it for building, only at runtime. I'm guessing this is erroring out when rpm-ostree runs dracut and tries to install the kernel module to the initramfs?

**Comment by jcajka** [comment]
@ajeddeloh yes

**Comment by jcajka** [comment]
Do you prefer some change to this repo or just "per package" patch in the spec?

**Comment by dustymabe** [comment]
right I was thinking it was probably `ignition-dracut`.. so we would make the change there:

https://github.com/coreos/ignition-dracut/blob/master/dracut/30ignition/module-setup.sh#L78

Can we do arch specific things in dracut modules ?

**Comment by ajeddeloh** [comment]
There's not a lot to change in this repo other than adding the ability to exclude platforms from the build process. I'm fine with building Ignition with qemu support that doesn't work on some arches short term. But long term we need to fix it or have a way to disable platforms at compile time. I suppose we could have a patch applied to the spec file that removes qemu support on unsupported platforms, but that seems hacky; it'd be better to make the ignition build process more modular.

The changes would probably need to be made around here: https://github.com/coreos/ignition-dracut/blob/master/dracut/30ignition/module-setup.sh#L78 but I'm not sure what the normal approach for specifying different behavior with different arches for dracut is.

**Comment by dustymabe** [comment]
i guess ignition-dracut is just a bash script so can probably just `uname -r` and act differently based on that.

**Comment by jcajka** [comment]
@ajeddeloh tbh me neither. I will dig in to the possibility to do it "cleanly" in dracut, otherwise IMHO some ifarch patch will be good for the mean time.

**Comment by jlebon** [comment]
@jaypoulz @Prashanth684 Has a channel for PPC64 been investigated and discussed?

**Comment by Prashanth684** [comment]
> @jaypoulz @Prashanth684 Has a channel for PPC64 been investigated and discussed?

No. it has been a more urgent requirement for s390x, becauseOpenshft libvirt IPI was the only way possible for running the CI. For ppc64, we have the Openstack platform for running the CI, so in terms of priority it has taken a backseat.
